builder/test/pipeline.cc

```diff
@@ -252,6 +252,73 @@ TEST_P(elementwise, pipeline_2d) {
   }
 }
 
+// An example of two 2D elementwise operations in sequence with intermediate buffer stored at
+// the inner loop level.
+TEST(elementwise, store_at) {
+
+  // Make the pipeline
+  node_context ctx;
+
+  auto in = buffer_expr::make(ctx, "in", 2, sizeof(int));
+  auto out = buffer_expr::make(ctx, "out", 2, sizeof(int));
+  auto intm1 = buffer_expr::make(ctx, "intm1", 2, sizeof(int));
+  auto intm2 = buffer_expr::make(ctx, "intm2", 2, sizeof(int));
+  auto intm3 = buffer_expr::make(ctx, "intm3", 2, sizeof(int));
+  auto intm4 = buffer_expr::make(ctx, "intm4", 2, sizeof(int));
+
+  // If we want to alias intermediate buffer to the output buffer,
+  // we need to tell aliaser that output is unfolded and it's safe to alias.
+  out->dim(0).fold_factor = dim::unfolded;
+  out->dim(1).fold_factor = dim::unfolded;
+
+  var x(ctx, "x");
+  var y(ctx, "y");
+
+  func mul1 = func::make(multiply_2<int>, {{in, {point(x), point(y)}}}, {{intm1, {x, y}}});
```

Comment:
Is there an existing test we could use for this? Maybe adding a store_at to https://github.com/dsharlet/slinky/blob/main/builder/test/elementwise.cc would hit the same issue?

Reply:
I think there are definitely tests which can be extended to do the same, but I'd really like to have at least one specific test which does just that.